src/tools/rados/rados.cc

```diff
@@ -239,6 +245,172 @@ void usage(ostream& out)
 "    --omap-key-file file            read the omap key from a file\n";
 }
 
+enum class UseStriper { yes, no };
```

Comment:
https://github.com/ceph/ceph/pull/21983#discussion_r189761082

Reply:
This idiom's worth explaining. Every year, I get closer to all but 
entirely avoiding the type "bool", especially as function parameter. The 
reasoning is that by making the type ontological a reader needs to know 
less about what the flag is. At the call site:
        f(true);
...compared to:
        f(laundry::fold_neatly);

It's a longer discussion, but that's the central idea. Going to the 
function's prototype is certainly a natural intuition, but there's no 
guarantee the names there will be well-chosen (I've seen examples in Ceph 
of Boolean parameters named "r" and "ld"), and this idiom doesn't save you 
from that. It does let you see something more specifically descriptive 
directly at the call site, though.

In this case, there's only one flag, so it might indeed be overkill-- but 
the choice here certainly wasn't a random one. I hope that helps to 
clarify things!

Reply:
it helps. but i still don't think we should perfer `UseStriper::yes` over a boolean parameter named `use_striper`. a bad parameter name of a boolean is indeed bad. but that's not a reason we should abandon boolean parameter.

for instance, what's your counteraction to fix an integer parameter named `r`? if we have a handful of bugs in a program written in C++, shall we rewrite it in Rust or Haskell?

Reply:
Correct, and bad parameter names aren't the primary reason I gave.

I feel I've explained the central reason why a "bool" parameter was avoided in favor of a strongly typed enumeration, which is definitely supported by C++.

While I certainly appreciate that it's not the way you would have written this (though I'm certainly not the first to lean in this direction-- consider LLVM's ForDefinition_t, for instance), if there's no actual problem, confusion, or consternation, I hope we can progress.

Reply:
i am not convinced.

Reply:
in general i agree with @chardan, and i'd prefer strong types in public interfaces. but these are private helper functions in a utility, and nothing ever calls them with `true` or `false` directly. there's a single call site that sets `use_striper = true;`, and the variable `use_striper` is used everywhere else. i don't think `enum UseStriper` has any advantage over bool in this case

Reply:
One observation is that single flags tend to accumulate friends over time, suggesting that it may be worthwhile to begin with making things explicit. This extends easily to interfaces to be sure, but but being explicit from the beginning means avoiding that whole line of evolution from the start.

In any case, I hope this can be discussed further in another forum. For now, let's focus on moving this feature forward: I've changed the flag back to bool.